internal/gengapic/gengapic.go

```diff
@@ -281,98 +281,102 @@ func (g *generator) getFormattedValue(m *descriptor.MethodDescriptorProto, field
 	return value, nil
 }
 
-func (g *generator) insertImplicitRequestHeaders(m *descriptor.MethodDescriptorProto) error {
-	headers, err := parseImplicitRequestHeaders(m)
-	if err != nil {
-		return err
+func (g *generator) insertImplicitRequestHeaders(m *descriptor.MethodDescriptorProto, headers [][]string) error {
+	if len(headers) == 0 {
+		return nil
 	}
-	if len(headers) > 0 {
```

Comment:
I didn't change this code, just de-dented it by checking the opposite case and returning early.